samples/apps/bookinfo/cluster_extension.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+#
+# Copyright 2017 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Script to run on a VM with VPC connectivity to the cluster, to prepare the DB and details app.
+
+
+# Initialize mysql for the rating demo
+function istioMysql() {
+  # The ratings script is using root/password by default.
+
+  cat <<EOF | sudo mysql
+GRANT ALL PRIVILEGES on *.* to 'root'@'localhost' IDENTIFIED BY 'password';
+FLUSH PRIVILEGES;
+CREATE DATABASE test;
+EOF
+
+  # Create the tables
+  mysql -u root -ppassword < mysql/mysqldb-init.sql
+}
+
+function istioUpdateRatings() {
+  local RATING=${1-2}
+
+  cat <<EOF | mysql -u root -ppassword
+  USE test;
+  update ratings set rating=$RATING where reviewid=1;
+EOF
+}
+
+function istioInstallBookinfo() {
+  # Configure the inbound ports
+  sudo bash -c 'echo "ISTIO_INBOUND_PORTS=80,9080,3306,27017" > /var/lib/istio/envoy/sidecar.env'
+  sudo chown istio-proxy /var/lib/istio/envoy/sidecar.env
+
+  sudo apt-get install -y ruby mariadb-server mongodb
+
+  sudo systemctl start mariadb
+  sudo systemctl restart istio
+  istioMysql
+
+  startDetails
+}
+
+function startDetails() {
+  # Start bookinfo components
+  local app=details
+  if [[ -r $app.pid ]] ; then
+     kill -9 $(cat ${app}.pid)
+  fi
+  ruby details/details.rb 9080  > details.log 2>&1 &
+  echo $! > $app.pid
+}
```

Comment:
Now this I am really confused. I thought your initial PRs were just testing. It kind of distorts the story if we say we are running in a kube pod, but let's take same app and run in a vm. The story is much more powerful, streamlined and memorable if we just say, assume you now have to talk to legacy db vm or a legacy service that you have not seen yet so far in the example flow.

Reply:
Sorry for the confusion - this is not for testing, but to allow people to see that some services can run on the VM, and you can have a 'mesh' of services that may include components running on-prem. 

I don't mind removing (or not showing) the details on the VM, AFAIK it was part of the demo use case - but I agree the DB is more compelling. 

Note that 'rating' won't work on the VM, since we don't yet support labels (like version)

Reply:
Its still not too late. We can whip up a simple Go/python based price service that talks http to bookinfo, and its a very small change in the UI to show price (fits nicely with book "info"). This service can talk to the DB, and let it run in the VM itself